lib/src/json_serializable/type_helper.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+abstract class TypeHelper {
+  const TypeHelper();
+
+  bool matchesTypeForSerialize(DartType type);
```

Comment:
I would unify these names, either change this to `canSerialize` or change `canDeserialize` to `matchesTypeForDeserialize`.